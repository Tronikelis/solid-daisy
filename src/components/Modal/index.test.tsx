import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it, vi } from "vitest";

import { Modal } from ".";

describe("<Modal />", () => {
    it("renders correctly", () => {
        const { container } = render(() => (
            <Modal open setOpen={vi.fn()}>
                oh
            </Modal>
        ));

        expect(container).toMatchSnapshot();
    });

    it("opening / closing works", async () => {
        const [open, setOpen] = createSignal(false);

        const screen = render(() => (
            <Modal open={open()} setOpen={setOpen}>
                foo
            </Modal>
        ));

        expect(document.querySelector("input")).not.toBeChecked();

        setOpen(true);

        expect(document.querySelector("input")).toBeChecked();

        await userEvent.click(screen.getByText("âœ•"));

        expect(document.querySelector("input")).not.toBeChecked();
    });
});
