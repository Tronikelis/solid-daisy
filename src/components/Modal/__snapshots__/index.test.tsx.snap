// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Modal /> > renders correctly 1`] = `
<div>
  <input
    class="modal-toggle"
    id="cl-0"
    type="checkbox"
  />
  <div
    class="modal"
  >
    <div
      class="modal-box"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="flex flex-row items-center gap-4 justify-end"
        >
          <button
            class="btn btn-sm btn-circle btn-ghost"
            type="button"
          >
            
            
            âœ•
            
          </button>
        </div>
        oh
      </div>
    </div>
  </div>
</div>
`;
